### PR TITLE
Update the WASM version to the latest xiplot

### PR DESCRIPTION
**Pull request by Aggrathon**
And bundle the `xiplot_filetypes` as a recommended plugin.

And create a GitHub action for verifying the build process on pull requests.

### CODE REVIEW

File: .github/workflows/test-build.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          submodules: true
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "3.x"
+
+      - name: Build the WASM version
+        run: |
+          pip install --upgrade pip
+          make deploy
```
Comment (juntyr) [inline .github/workflows/test-build.yaml:23]: Should we use `make -B deploy` here as well, as in the build CI?

**Review by juntyr — APPROVED** [review]
LGTM!

File: .github/workflows/test-build.yaml (hunk 1 above)
Comment (Aggrathon) [reply .github/workflows/test-build.yaml:23]: I'll add it

Comment (Aggrathon) [reply .github/workflows/test-build.yaml:23]: Would it be okay if I add a `<h1>&chi;iplot</h1>` to `index.html` (above "Initialising the WebDash environment")?

Comment (juntyr) [reply .github/workflows/test-build.yaml:23]: Yes, that’s absolutely ok

**Pull request merged**